website/src/documentation/programming-guide.md

```diff
@@ -1970,7 +1976,1076 @@ records.apply("WriteToText",
+  public abstract String getBank(); 
+  public abstract double getPurchaseAmount();
+}
+```
+
+This is all that’s needed to generate a simple AutoValue class, and the above `@DefaultSchema` annotation tells Beam to
+infer a schema from it. This also allows AutoValue elements to be used inside of `PCollection`s.
+
+`@SchemaFieldName` and `@SchemaIgnore` can be used to alter the schema inferred.
+
+### 6.6. Using Schemas {#using-schemas}
+A schema on a `PCollection` enables a rich variety of relational transforms. The fact that each record is composed of
+named fields allows for simple and readable aggregations that reference fields by name, similar to the aggregations in
+a SQL expression. 
+
+#### 6.6.1. Field selection syntax
+The advantage of schemas is that they allow referencing of element fields by name. Beam provides a selection syntax for
+referencing fields, including nested and repeated fields. This syntax is used by all of the schema transforms when 
+referencing the fields they operate on. The syntax can also be used inside of a DoFn to specify which schema fields to
+process.
+
+Addressing fields by name still retains type safety as Beam will check that schemas match at the time the pipeline graph
+is constructed. If a field is specified that does not exist in the schema, the pipeline will fail to launch. In addition,
+if a field is specified with a type that does not match the type of that field in the schema, the pipeline will fail to
+launch.
+
+##### **Top-level fields**
+In order to select a field at the top level of a schema, the name of the field is specified. For example, to select just
+the user ids from a `PCollection` of purchases one would write (using the `Select` transform)
+
+```java
+purchases.apply(Select.fieldNames(“userId”));
+```
+
+##### **Nested fields**
+Individual nested fields can be specified using the dot operator. For example, to select just the postal code from the
+ shipping address one would write
+
+```java
+purchases.apply(Select.fieldNames(“shippingAddress.postCode”));
+```
+
+##### **Wildcards**
+The * operator can be specified at any nesting level to represent all fields at that level. For example, to select all
+shipping-address fields one would write
+
+```java
+purchases.apply(Select.fieldNames(“shippingAddress.*”));
+```
+
+##### **Arrays**
+An array field, where the array element type is a row, can also have subfields of the element type addressed. When 
+selected, the result is an array of the selected subfield type. For example
+
+```java
+purchases.apply(Select.fieldNames(“transactions[].bank”));
+```
+
+Will result in a row containing an array field with element-type string, containing the list of banks for each 
+transaction. 
+
+While the use of  [] brackets in the selector is recommended, to make it clear that array elements are being selected, 
+they can be omitted for brevity. In the future, array slicing will be supported, allowing selection of portions of the 
+array.
```

Comment:
What do you think about linking JIRAs for these "in the future" items?

Reply:
unless there's precedent for linking JIRAs in the docs, I'd rather not.